pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -636,23 +654,31 @@ def build_package(
     name = pkg["package"]["name"]
     build_dir = pkg_root / "build"
     src_dir_name: str = name + "-" + pkg["package"]["version"]
-    src_path = build_dir / src_dir_name
+    srcpath = build_dir / src_dir_name
     source_metadata = pkg["source"]
     build_metadata = pkg["build"]
+
+    if not force and not needs_rebuild(pkg_root, build_dir, source_metadata):
+        return
+
     with chdir(pkg_root), get_bash_runner() as bash_runner:
-        if not needs_rebuild(pkg_root, build_dir, source_metadata):
-            return
-        if source_metadata:
-            if build_dir.resolve().is_dir():
-                shutil.rmtree(build_dir)
-            os.makedirs(build_dir)
-
-        srcpath = prepare_source(pkg_root, build_dir, src_path, source_metadata)
-        if build_metadata.get("script"):
-            run_script(build_dir, srcpath, build_metadata, bash_runner)
-        if build_metadata.get("library"):
-            create_packaged_token(build_dir)
-            return
+        if not should_prepare_source:
+            assert srcpath.exists()
```

Comment:
Let's raise an AssertionError or a IOError, with a verbose error message that includes the path. Otherwise this will be hard to debug when it fails.